src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -486,6 +488,17 @@ public String getName() {
         return name;
     }
 
+    /**
+     * {@return the string representation of the ZIP file}
+     *
+     * This string is equivalent to the base name in the value returned by
+     * {@link #getName()}.
+     */
+    @Override
+    public String toString() {
```

Comment:
I don't think this is too helpful as the file name may not be unique or there may be several instances of ZipFile that are backed by the same zip file. Can you try `"" + file + Integer.toHexString(System.identityHashCode(file))` ?

Reply:
Hi Alan,

Thanks for taking a look. I updated the toString() value to the one you suggested, and also dropped the specific aspects of the specification. 

I am not sure if you have a preference one way or another regarding providing the full path versus just the file name, but I can switch the full path for just the file name if need be.

Reply:
I like the new wording (have no oppinion if absolute path is better).

Reply:
Hello Justin,

> I am not sure if you have a preference one way or another regarding providing the full path versus just the file name, but I can switch the full path for just the file name if need be.

My opinion is that we should not use the absolute path here. Section 2.1 of secure coding guidelines https://www.oracle.com/java/technologies/javase/seccodeguide.html#2-1 suggests not to include full paths in exception messages.

With the proposed change to the toString() method here, which uses absolute paths, I think it would then mean that we would have to review (within the JDK) usages of (explicit or implicit) `ZipFile.toString()` to prevent accidentally including the complete paths in the exceptions, like in the case below:

```java
final ZipFile zf = new ZipFile("/home/me/xyz.zip");
... // do something 
throw new Exception("failed to handle zipfile " + zf);
```

Reply:
Thanks Jai, that makes sense. Replaced full path with just the base name in latest commit.

Reply:
I think second paragraph of the method description needs to be dropped. Documenting the representation and then saying it is subject to change is going to cause problems now the road.

Reply:
I agree we should probably remove the 2nd paragraph